src/services/express_request.rs

```diff
@@ -1,260 +1,262 @@
+where
+    S: serde::Serializer,
+{
+    let date = date.with_timezone(&Local);
+    let s = date.format("%Y%m%d%H%M%S").to_string();
+    serializer.serialize_str(&s)
 }
 
+// TODO:: The success response has more fields than this
 #[derive(Debug, Clone, Deserialize)]
-pub struct MpesaExpressRequestResponse {
-    #[serde(rename(deserialize = "CheckoutRequestID"))]
+#[serde(rename_all = "PascalCase")]
+pub struct MpesaExpressResponse {
+    ///This is a global unique identifier of the processed checkout transaction
+    /// request.
+    #[serde(rename = "CheckoutRequestID")]
     pub checkout_request_id: String,
-    #[serde(rename(deserialize = "CustomerMessage"))]
+    /// This is a message that your system can display to the customer as an
+    /// acknowledgment of the payment request submission.
     pub customer_message: String,
-    #[serde(rename(deserialize = "MerchantRequestID"))]
+    /// This is a global unique Identifier for any submitted payment request.
+    #[serde(rename = "MerchantRequestID")]
     pub merchant_request_id: String,
-    #[serde(rename(deserialize = "ResponseCode"))]
+    /// This is a Numeric status code that indicates the status of the
+    /// transaction submission. 0 means successful submission and any other
+    /// code means an error occurred.
     pub response_code: String,
-    #[serde(rename(deserialize = "ResponseDescription"))]
+    ///Response description is an acknowledgment message from the API that
+    /// gives the status of the request submission. It usually maps to a
+    /// specific ResponseCode value.
+    ///
+    /// It can be a Success submission message or an error description.
     pub response_description: String,
 }
 
-pub struct MpesaExpressRequestBuilder<'mpesa, Env: ApiEnvironment> {
-    business_short_code: &'mpesa str,
+#[derive(Builder, Debug, Clone)]
+#[builder(build_fn(error = "MpesaError", validate = "Self::validate"))]
+pub struct MpesaExpress<'mpesa, Env: ApiEnvironment> {
+    #[builder(pattern = "immutable")]
     client: &'mpesa Mpesa<Env>,
-    transaction_type: Option<CommandId>,
-    amount: Option<f64>,
-    party_a: Option<&'mpesa str>,
-    party_b: Option<&'mpesa str>,
-    phone_number: Option<&'mpesa str>,
-    callback_url: Option<&'mpesa str>,
-    account_ref: Option<&'mpesa str>,
+    /// This is the organization's shortcode (Paybill or Buygoods - A 5 to
+    /// 6-digit account number) used to identify an organization and receive
+    /// the transaction.
+    #[builder(setter(into))]
+    business_short_code: &'mpesa str,
+    /// This is the transaction type that is used to identify the transaction
+    /// when sending the request to M-PESA
+    ///
+    /// The TransactionType for Mpesa Express is either
+    /// `CommandId::BusinessBuyGoods` or
+    /// `CommandId::CustomerPayBillOnline`
+    transaction_type: CommandId,
+    /// This is the Amount transacted normally a numeric value
+    #[builder(setter(into))]
+    amount: f64,
```

Comment:
Same to this, floating point numbers rather than `<num>.0` are not accepted. We might as well use a `u32` here to avoid validation errors from safaricom